### ISSUE TITLE
Make webhooks utilities available as a static property

### ISSUE THREAD

**Issue by brettwillis** [stripe/stripe-node#1956]
### Is your feature request related to a problem? Please describe.

In some scenarios for me (normally when using Stripe Connect), we don't have enough information to create a Stripe instance until we have parsed the webhook event.

Specifically, we may need the `Event.account` property to determine what credentials/API key to use to create a Stripe instance.

However, the `Stripe.Webhooks` utility is only available *after* creating a Stripe instance.

### Describe the solution you'd like

After inspecting the code, it seems the webhook utilities instance does not depend at all on a specific Stripe instance.

https://github.com/stripe/stripe-node/blob/12fe437bb94bb088a3725d79292979cbeacb3302/src/Webhooks.ts#L70-L72

This webhook utilites instance could be made available as a static property on the `Stripe` constructor, and that instance also shared on the individual instances.

Like so:

```js
const Stripe = require('stripe');

// Available as a static property
Stripe.webhooks.constructEventAsync(...);

// Remains available on the instance also
const client = Stripe('...');
client.webhooks === Stripe.webhooks // true
```

Happy to help with a PR if you agree with the logic.

### Describe alternatives you've considered

A workaround is using `new Stripe('').webhooks` to get an instance of the webhook utilities, but it feels like a hack, because it relies on the internal workings of the `Stripe()` constructor not attempting to validate the empty API key.

**Comment by brettwillis** [comment]
Ah I just noticed that `Stripe.webhooks` is already assigned as the factory function.
https://github.com/stripe/stripe-node/blob/12fe437bb94bb088a3725d79292979cbeacb3302/src/stripe.core.ts#L158

I don't know if this would make it a breaking change or not. It doesn't seem to actually document that property in the typings?

The problem with this factory function is that it requires an instance of `platformFunctions` which is also not easily available.

### Alternative 1

Maybe an alterative is to make the `Stripe.webooks` factory function have the `platformFunctions` parameter default to the `platformFunctions` used by the current Stripe object. Like

```js
// Stripe.webhooks = createWebhooks --- change this
Stripe.webhooks = (fns) => createWebhooks(fns ?? platformFunctions);
``` 

While this is not as elegant as single instance that can be shared around Stripe instances, it's a step in the right direction.

### Alternative 2

Otherwise we change the name `Stripe.webhooksInstance` or something, I don't know if I'd call that elegant either.

### Alternative 3 ✅ 

Maintain compatibillity and incorporate both proposed improvements like this

```js
Stripe.webhooks = Object.assign(
  (fns) => createWebhooks(fns ?? platformFunctions),
  createWebhooks(platformFunctions),
);
```

So that `Stripe.webhooks` is both a factory function with a default parameter and a valid webhooks utilities instance.

**Comment by anniel-stripe** [comment]
Hi @brettwillis , thanks for the detailed feature request! I agree this is something that we should fix to better support Connect workflows.

I dug into this more, and it seems like `Stripe.webhooks` [used to be a reference](https://github.com/stripe/stripe-node/blob/70c626ec981945ff13c70785f38c40c884beec71/src/Webhooks.ts#L355) to an instance of the `WebhookObject` utility, but we unintentionally broke this interface when we refactored the library to split out platform-specific logic ([specifically this PR](https://github.com/stripe/stripe-node/pull/1679)). I believe the disciplined fix here is to create the webhook utilities instance on `Stripe`:
```javascript
Stripe.webhooks = createWebhooks(platformFunctions)  // this gets reused on the Stripe client instance
```
since you're right that it doesn't depend on the Stripe instance. This will be a breaking change though, and we would do this in the next major version.

In the meantime I do think Alternative 3 would be right here. Happy to accept a PR for Alternative 3 if you have the time, or I can take it. If you do open a PR, it would be nice to have more explicit code than `Object.assign`, or at least include a descriptive comment.

**Comment by brettwillis** [comment]
Ok, great, I'll submit a PR shortly.

> it would be nice to have more explicit code than `Object.assign`

Not exactly sure what you mean by more explicit, but yes I'll definitely explain it in comments.

**Comment by anniel-stripe** [comment]
Thanks for the awesome PR! I'll close this issue once the changes are included in this Thursday's release.

**Comment by anniel-stripe** [comment]
Released in [v14.5.0](https://github.com/stripe/stripe-node/releases/tag/v14.5.0).
